### ISSUE TITLE
False positive TC004

### ISSUE THREAD

**Issue by coredumperror** [snok/flake8-type-checking#205]
Using the latest Ruff, which uses this plugin, I'm getting a false positive detection of [TC004](https://docs.astral.sh/ruff/rules/runtime-import-in-type-checking-block/#runtime-import-in-type-checking-block-tc004) with this code:

```
if TYPE_CHECKING:
    from django_stubs_ext import StrOrPromise

...

class ApplicationTable(BasicModelTable):
    ...

    #: The URL to submit form actions to
    form_url: StrOrPromise = reverse_lazy("operations:applications--actions")

```

This seems like a bug to me, because StrOrPromise is only used in type-hints... so why is it triggering TC004? Wrapping `StrOrPromise` in quotes makes the TC0004 error go away... but why should that be needed?
